client/src/partials/templates/bhFindDocument.tmpl.html

```diff
@@ -1,156 +1,130 @@
+              <a class="media-left">
+                <i class="fa fa-3x" ng-class="{{ '$ctrl.mimeIcon(doc.mimetype).icon' }}"></i>
+              </a>
+              <div class="media-body">
+
+                <!-- actions buttons  -->
+                <div class="pull-right">
+                  <a
+                    ng-href="/{{ doc.link }}" download="{{ doc.downloadLink }}"
+                    title="{{ 'PATIENT_DOCUMENT.DOWNLOAD' | translate }}"
+                    type="button"
+                    class="btn btn-sm btn-default">
+                    <i class="fa fa-download"></i>
+                  </a>
+
+                  <button
+                    ng-click="$ctrl.deleteDocument(doc.uuid)"
+                    title="{{ 'FORM.BUTTONS.DELETE' | translate }}"
+                    type="button"
+                    class="btn btn-sm btn-danger">
+                    <i class="fa fa-trash"></i>
+                  </button>
                 </div>
-              </div>
-              <div class="document-lable">
-                <b class="btn-sm pull-right">
-                  <i ng-show="$ctrl.session.showAction && $ctrl.selectedIndex === $index" class="fa fa-angle-up"></i>
-                  <i ng-hide="$ctrl.session.showAction && $ctrl.selectedIndex === $index" class="fa fa-angle-down"></i>
-                </b>
 
+                <!-- title  -->
                 <h4 class="media-heading">{{ doc.label }}</h4>
 
-                <div ng-show="$ctrl.session.showAction && $ctrl.selectedIndex === $index">
-                  <i class="fa fa-file-o"></i>
-                  {{ $ctrl.mimeIcon(doc.mimetype).label }} (~{{ doc.size | bytes }})
-                </div>
+                <!-- file size  -->
+                <i class="fa fa-file-o"></i>
+                {{ $ctrl.mimeIcon(doc.mimetype).label }} (~{{ doc.size | bytes }}) |
 
+                <!-- date and author  -->
                 <i class="fa fa-calendar-o"></i>
                 {{ doc.timestamp | date }}, {{ 'FORM.LABELS.BY' | translate }}
                 {{ doc.first + ' ' + doc.last }}
               </div>
-              <div ng-show="$ctrl.session.showAction && $ctrl.selectedIndex === $index" class="document-actions">
-                <button
-                  title="{{ 'PATIENT_DOCUMENT.OVERVIEW' | translate }}"
-                  type="button"
-                  class="btn btn-sm btn-default">
-                  <i class="fa fa-eye"></i>
-                </button>
+            </div>
+          </div>
+      </div>
+
+      <!-- thumbnail display mode  -->
+      <div ng-switch-when="thumbnail">
+        <div class="document document-thumbnail col-xs-12 col-md-6"
+            ng-repeat="doc in $ctrl.session.patientDocuments | filter: $ctrl.searchByName">
+
+            <!-- the picture  -->
+            <div class="crop">
+              <div class="image">
+                <img ng-show="$ctrl.mimeIcon(doc.mimetype).icon === 'fa-file-image-o'" ng-src="{{ doc.link }}" />
```

Comment:
In the future, it would be good to keep functions out of the view.  If possible, it would be better to write:
```js
// in the controller, possibly on image download
doc.icon = mimeIcon(doc.mimetype).icon;
```
... in the HTML ...
```html
<img ng-show="doc.icon === 'fa-file-image-o'" ...></img>
```